.github/workflows/deploy_docs.yml

```diff
@@ -0,0 +1,60 @@
+name: Deploy Documentation
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*'
+  workflow_dispatch:
+
+# Only allow one docs build at a time so that overlapping stale builds will get
+# cancelled automatically.
+concurrency:
+  group: deploy_docs
+  cancel-in-progress: true
+
+jobs:
+  build-and-deploy:
+    name: Build & Deploy
+    runs-on: ubuntu-latest
+
+    permissions:
```

Comment:
I'm not entirely sure if we need all three of these write permissions. I was mix-and-matching from a few different examples, so it's possible they're not all needed now.

Reply:
Update: yes, we absolutely need all of these permissions here.